### ISSUE TITLE
Unexpected performance loss of Zero-shot on ScanObjectNN.

### ISSUE THREAD

**Issue by Mr-Neko** [salesforce/ULIP#21]
Hi! Thanks for your contribution to open source. I try to make a scanobjectnn dataset from point-bert as you said, where the code is:
```python
    @DATASETS.register_module()
    class ScanObjectNN(data.Dataset):
        def __init__(self, config):
            super().__init__()
            self.root = config.DATA_PATH
    
            h5 = h5py.File(os.path.join(self.root, 'main_split', 'test_objectdataset_augmentedrot_scale75.h5'), 'r')
            self.points = np.array(h5['data']).astype(np.float32)
            self.labels = np.array(h5['label']).astype(int)
            h5.close()
    
            with open(os.path.join(self.root, 'ScanObjectNN_shape_names.txt'), 'r') as f:
                lines = f.readlines()
                lines = [line.rstrip() for line in lines]
            
            self.shape_name=lines
    
    
        def __getitem__(self, idx):
            pt_idxs = np.arange(0, self.points.shape[1])   # 2048
            
            current_points = self.points[idx, pt_idxs].copy()
            
            # current_points = pc_normalize(current_points)
            current_points = torch.from_numpy(current_points).float()
    
            label = self.labels[idx]
            
            return current_points, label, self.shape_name[int(label)]
    
        def __len__(self):
            return self.points.shape[0]
```
I use your checkpoint to test it. However, I got 15 percent loss as the top 1 accuracy is 30.3. I wonder that is there any process is wrong? Looking forward your reply!!

**Comment by Tycho-Xue** [comment]
Hi @Mr-Neko, For fine-tune on ScanobjectNN dataset, we used hardest mode following the common practice; but for zero-shot classification on ScanobjectNN dataset, we use the obj-only mode, for the following reasons: 1. In the pre-training, we only have the examples of obj-only type of objects for 3D point clouds, and so as the rendered images. 2. We compare to PointCLIP for the zero-shot classification, it's a bit unclear what mode of ScanobjectNN test dataset they used, since this part of the code is not released, according to their paper, the test set is 581 samples, which matches the obj-only test set statistics(the hardest mode has more than 2k samples since it's augmented.), and no matter what mode to compare, we still beat pointclip by a significant margin.
I hope this addresses your question, thanks for your interest in our work.

**Comment by Mr-Neko** [comment]
Thanks！It's clear. By the way, could you provide the hyper parameters for finetune pointmlp on modelnet40 and scanobjectnn? I tried your checkpoint to reproduce the fine tune results, but it's hard to train the acc as high as paper metioned. Thx a lot.

**Comment by kaimingkuang** [comment]
@Tycho-Xue @Mr-Neko I modified the ModelNet zero-shot pipeline for ScanObjectNN and got similar results on the OBJ_ONLY variation (33.2% top-1 accuracy). Is there anything that is specific for ScanObjectNN missing from the ModelNet pipeline (e.g., orientation, point-cloud normalization)?

**Comment by Mr-Neko** [comment]
Nope, I test the OBJ_ONLY without any difference with the ModelNet, and the results are normal. @kaimingkuang
